src/Spectre.Data.Tests/RoiUtilitiesTests.cs

```diff
@@ -0,0 +1,110 @@
+    {
+        private readonly string _path = TestContext.CurrentContext.TestDirectory + "\\..\\..\\..\\..\\..\\test_files\\Rois";
+        private string _testpath;
+        private string _testfilespath;
+        private readonly RoiDataset _readroidataset = new RoiDataset();
+        private readonly RoiDataset _writeroidataset = new RoiDataset();
+
+        [SetUp]
+        public void SetUp()
+        {
+            _testpath = Path.GetFullPath(_path);
+            _testfilespath = _testpath + "\\image1.png";
+            _readroidataset.Name = "image1";
+            _readroidataset.Height = 6;
+            _readroidataset.Width = 6;
+            _readroidataset.RoiPixels = new List<RoiPixel>
+            {
+                new RoiPixel(1, 1),
+                new RoiPixel(2, 1),
+                new RoiPixel(3, 1)
+            };
+
+            _writeroidataset.Name = "writetestfile";
+            _writeroidataset.Height = 10;
+            _writeroidataset.Width = 10;
+            _writeroidataset.RoiPixels = new List<RoiPixel>
+            {
+                new RoiPixel(1, 5),
+                new RoiPixel(2, 5),
+                new RoiPixel(3, 5),
+                new RoiPixel(4, 5),
+            };
+        }
+
+        [Test]
+        public void ListRoisFromDirectory_returns_proper_names()
+        {
+            RoiUtilities service = new RoiUtilities(_testpath);
+
+            var names = service.ListRoisFromDirectory();
+
+            Assert.AreEqual("image1.png", Path.GetFileName(names[0]));
+            Assert.AreEqual("image2.png", Path.GetFileName(names[1]));
+        }
+
+        [Test]
+        public void ReadRoi_returns_proper_roi_pixels()
+        {
+            RoiUtilities service = new RoiUtilities(_testfilespath);
+            var roi = service.RoiReader();
+
+            Assert.AreEqual(roi.RoiPixels[0].GetXCoord(), _readroidataset.RoiPixels[0].GetXCoord());
+            Assert.AreEqual(roi.RoiPixels[0].GetYCoord(), _readroidataset.RoiPixels[0].GetYCoord());
+
+            Assert.AreEqual(roi.RoiPixels[1].GetXCoord(), _readroidataset.RoiPixels[1].GetXCoord());
+            Assert.AreEqual(roi.RoiPixels[1].GetYCoord(), _readroidataset.RoiPixels[1].GetYCoord());
+
+            Assert.AreEqual(roi.RoiPixels[2].GetXCoord(), _readroidataset.RoiPixels[2].GetXCoord());
+            Assert.AreEqual(roi.RoiPixels[2].GetYCoord(), _readroidataset.RoiPixels[2].GetYCoord());
+        }
+
+        [Test]
+        public void ReadRoi_returns_proper_dimensions()
+        {
+            RoiUtilities service = new RoiUtilities(_testfilespath);
+            var roi = service.RoiReader();
+
+            Assert.AreEqual(roi.Height, _readroidataset.Height);
+            Assert.AreEqual(roi.Width, _readroidataset.Width);
+        }
+
+        [Test]
+        public void WriteRoi_writes_file_properly()
+        {
+            RoiUtilities service = new RoiUtilities(_testpath);
+
+            service.RoiWriter(_writeroidataset);
```

Comment:
You don't check, whether it has been properly written - only that this operation does not throw. It would be useful to read it back and check integrity.